test/test_dynamic_shapes.py

```diff
@@ -714,6 +715,73 @@ def test_method(self, fn, first_type, second_type):
 
             self._do_test(fn, inp1, inp2, shape_env, is_unary_fn)
 
+    def test_non_symbolic_symnode(self):
+        j1 = torch._C._get_singleton_int(1)
+        j2 = torch._C._get_singleton_int(1)
+        j3 = torch._C._get_singleton_int(3)
+
+        self.assertIsInstance(j1, torch.SymInt)
+        self.assertNotIsInstance(j1, int)
+
+        with self.assertRaisesRegex(RuntimeError, "add not supported by SingletonSymNode"):
+            j1 + 3
+
+        self.assertFalse(j1 == 3)
+        self.assertFalse(3 >= j2)
+
+        def test_constant_symbool(x, val):
+            self.assertIsInstance(x, torch.SymBool)
+            self.assertFalse(is_symbolic(x))
+            self.assertIs(x.node.guard_bool("", 0), val)
+
+        test_constant_symbool(j1 == j1, True)
+        test_constant_symbool(j1 == j2, True)
+        test_constant_symbool(j1 == j3, False)
+        test_constant_symbool(j1 != j3, True)
+        test_constant_symbool(j1 != j2, False)
+
+        x = j1 == j2
```

Comment:
Yes, we can! It is updated now to do so. Originally I had preferred handling constant SymBools in the inputs vs the inputs, since I assumed that there are other ways to get constant SymBools to appear as inputs, but after looking again, it seems unlikely to see them in the wild anymore. I'm considering to remove all the logic to handle constant SymBools in the inputs once I make sure that they don't appear for more compilation test cases. But for now, I've added an alternative way of testing this logic that we currently have.